### PR TITLE
Private Team support added

### LINKED ISSUE #93: Private channel user invitation message
Currently Atomist does not support the bot inviting user into private channels. When trying to do so the client will error at which point we should catch the error and respond to the user asking them to manually invite the user.

### PR DESCRIPTION
**Pull request by kbristow**
- Fixed bug in adding new members to team.
- Made it possible for users to be added to gluon teams from private channels
- Manually inviting the atomist bot and users to private channels is still required but users are now made aware with a helpful message.
- Removed role param from AddTeamMember because it was not used

New command flow looks as follows when linking a private slack channel:
screen shot 2018-03-20 at 16 30 59

After inviting the atomist bot to the private channel:
screen shot 2018-03-20 at 16 31 15

Choosing to add new team members:
screen shot 2018-03-20 at 16 31 30

Result of adding a team member to a private slack channel linked gluon team:
screen shot 2018-03-20 at 16 31 43

Resolves #93. Relates to #43 but we will close that one when the atomist bot supports the invitation functionality we are working around here.

### CODE REVIEW

File: src/atomist.config.ts [hunk 1]
```diff
@@ -59,14 +61,13 @@ import {
...
-
 const token = QMConfig.token;
 
 export const configuration: any = {
-    name: pj.name,
-    version: pj.version,
     teamIds: [QMConfig.teamId],
+    // running durable will store and forward events when the client is disconnected
+    // this should only be used in production envs
+    policy: process.env.NODE_ENV === "production" ? "durable" : "ephemeral",
```
Comment (chriskieser) [inline src/atomist.config.ts]: Cool. Don was telling me about this today

Comment (kbristow) [reply src/atomist.config.ts]: @cdupuis added this for us. I was not aware of it but it looks cool.

File: src/gluon/team/TeamSlackChannel.ts [hunk 2]
```diff
@@ -219,54 +217,16 @@ function linkSlackChannelToGluonTeam(ctx: HandlerContext,
...
-                            }, error => {
-                                logger.error(`Error creating Slack channel: ${JSON.stringify(error)}`);
-
-                                if (error.networkError.response.status === 400) {
-                                    logger.warn(`Most likely trying to link a private Slack channel: ${error.message}. This is currently NOT SUPPORTED`);
-                                } else {
-                                    return Promise.reject(`Slack channel could not be created: ${error.message}`);
+                            }, err => {
+                                if (err.networkError.response.status === 400) {
+                                    return ctx.messageClient.respond(`The channel has been successfully linked to your team but the since that channel "${finalisedSlackChannelName}" is private` +
```
Comment (chriskieser) [inline src/gluon/team/TeamSlackChannel.ts]: ** but the since **

Comment (chriskieser) [reply src/gluon/team/TeamSlackChannel.ts]: Also need a space between 'private' and 'the'

Comment (kbristow) [reply src/gluon/team/TeamSlackChannel.ts]: Swear I fixed this. Probably decided to rewrite it again and then wrote it badly for the 10th time. Will fix.

**Pull request merged**